srcpkgs/base-files/INSTALL

```diff
@@ -4,58 +4,57 @@ make_system_dirs() {
 	#
 	for d in boot etc etc/modprobe.d etc/modules-load.d \
 		etc/skel home dev proc usr mnt opt sys media var run/lock; do
-		[ ! -d ${d} ] && install -d ${d}
+		mkdir -p ${d}
 	done
 
-	[ ! -d root ] && install -dm750 root
+	mkdir -p root && chmod 750 root
 
 	# Don't try to create var/mail in the correct place if the user
 	# is updating from an old system that has var/mail as a symlink
-	[ ! -L var/mail ] && [ ! -d var/mail ] && install -dm1777 var/mail
+	[ ! -L var/mail ] && mkdir -p var/mail && chmod 1777 var/mail
 
-	[ ! -d var/spool ] && install -d var/spool
+	mkdir -p var/spool
 
 	for d in local local/bin local/sbin local/include local/lib \
 		bin include lib src; do
-		[ ! -d usr/${d} ] && install -d usr/${d}
+		mkdir -p usr/${d}
 	done
 
 	for d in locale misc terminfo zoneinfo doc info; do
-		[ ! -d usr/share/${d} ] && install -d usr/share/${d}
-		[ ! -d usr/local/share/${d} ] && install -d usr/local/share/${d}
+		mkdir -p usr/share/${d}
+		mkdir -p usr/local/share/${d}
 	done
 
 	for d in 1 2 3 4 5 6 7 8; do
-		[ ! -d usr/share/man/man${d} ] && \
-			install -d usr/share/man/man${d}
-		[ ! -d usr/local/share/man/man${d} ] && \
-			install -d usr/local/share/man/man${d}
+		mkdir -p usr/share/man/man${d}
+		mkdir -p usr/local/share/man/man${d}
 	done
 
 	for d in empty log opt cache lib; do
-		[ ! -d var/${d} ] && install -d var/${d}
+		mkdir -p var/${d}
 	done
 
 	# Create /var/run and /var/lock symlinks.
 	for d in run lock; do
-		if [ ! -h "var/$d" -a -d var/${d} ]; then
+		if [ ! -h "var/$d" ] && [ -d var/${d} ]; then
 			echo "/${d} must not be a directory, exiting!"
 			exit 1
 		fi
 	done
 
-	cd var
-	ln -sf ../run .
-	ln -sf ../run/lock .
-	[ ! -d spool/mail ] && ln -sfn ../mail spool/mail
-	if [ -L spool/mail/mail -a "$(readlink spool/mail/mail)" = spool/mail ]; then
-		# Get rid of broken symlink created by older versions of base-files.
-		rm spool/mail/mail
-	fi
-	cd ..
+	(
```

Comment:
what's the point of wrapping this whole block?

Reply:
When `cd var` fails, we run `cd ..` which could go out of the chroot. I didn't observe this on a real system, but it sounds like a possibility.

Reply:
Limiting a PWD change to a subshell is always a good idea.